temporian/core/test/evaluation_test.py

```diff
@@ -113,6 +113,76 @@ def test_evaluate_dict(self):
         self.assertLen(result, 2)
         self.assertEqual(set(result.keys()), {"i1", "i2"})
 
+    def test_evaluate_named_input_data(self):
+        i1 = utils.create_input_node("i1")
+        i2 = utils.create_input_node("i2")
+
+        result = evaluation.evaluate(
+            [i1, i2],
+            {
+                "i1": utils.create_input_event_set(),
+                "i2": utils.create_input_event_set(),
+            },
+        )
+
+        self.assertIsInstance(result, list)
+        self.assertLen(result, 2)
+
+    def test_evaluate_mixed_input_data(self):
+        i1 = utils.create_input_node("i1")
+        i2 = utils.create_input_node("i2")
+
+        result = evaluation.evaluate(
+            [i1, i2],
+            {
+                "i1": utils.create_input_event_set(),
+                i2: utils.create_input_event_set(),
+            },
+        )
+
+        self.assertIsInstance(result, list)
+        self.assertLen(result, 2)
+
+    def test_evaluate_single_unnamed_input_no_name(self):
+        e1 = utils.create_input_event_set()
+        i1 = e1.node()
+
+        with self.assertRaises(ValueError):
+            evaluation.evaluate(i1, e1)
+
+    def test_evaluate_single_unnamed_input_named(self):
+        e1 = utils.create_input_event_set("i1")
+        i1 = e1.node()
+
+        result = evaluation.evaluate(i1, e1)
+
+        self.assertIsInstance(result, EventSet)
+        self.assertTrue(result is e1)
+
+    def test_evaluate_list_unnamed_inputs_no_name(self):
+        e1 = utils.create_input_event_set("i1")
+        i1 = e1.node()
+        e2 = utils.create_input_event_set()
+        i2 = e2.node()
+
+        with self.assertRaises(ValueError):
+            evaluation.evaluate([i1, i2], [e1, e2])
+
+    def test_evaluate_list_unnamed_inputs_named(self):
+        e1 = utils.create_input_event_set("i1")
+        i1 = e1.node()
+        e2 = utils.create_input_event_set("i2")
+        i2 = e2.node()
+        e3 = utils.create_input_event_set("i3")
+        i3 = e3.node()
+
+        result = evaluation.evaluate([i3, i1, i2], [e1, e2, e3])
+
+        self.assertIsInstance(result, list)
+        self.assertTrue(result[0] is e3)
+        self.assertTrue(result[1] is e1)
+        self.assertTrue(result[2] is e2)
+

```

Comment:
I'd add a test to check failure on duplicate node names

Reply:
added three